pkg/gadgets/seccomp/gadget.go

```diff
@@ -191,6 +184,103 @@ func seccompProfileAddLabelsAndAnnotations(
 	}
 }
 
+type SeccompProfileNsName struct {
+	namespace string
+	name      string
+
+	// generateName indicates whether the name field has to be used as
+	// resource's Name or GeneratedName
+	generateName bool
+}
+
+// getSeccompProfileNextName computes the next profile name that has to be used
+// for a specific podname given a SeccompProfile list. This function returns:
+// podName: If there do not exist profiles with podname or podname-X as name.
+// podName-2: If there exist a profile with the podname but no one with podname-X.
+// podName-<X+1>: If there exist at least one profile with podname-X.
+func getSeccompProfileNextName(profileList []seccompprofilev1alpha1.SeccompProfile, podName string) string {
+	currentCounter := 0
+	for _, profile := range profileList {
+		if !strings.HasPrefix(profile.Name, podName) {
+			continue
+		}
+
+		if profile.Name == podName && currentCounter == 0 {
+			currentCounter++
+			continue
+		}
+
+		c, err := strconv.Atoi(strings.TrimLeft(profile.Name, podName+"-"))
+		if err != nil {
+			// Ignore profiles with "podname-X" name where X is not a number.
+			continue
+		}
+
+		if c > currentCounter {
+			currentCounter = c
+		}
+	}
+
+	// It is the first profile for this pod, use the podname as resource's name.
+	if currentCounter == 0 {
+		return podName
+	}
+
+	return fmt.Sprintf("%s-%d", podName, currentCounter+1)
+}
+
+// getSeccompProfileNsName computes the seccomp profile namespace and name
+// based on the traceOutputName parameter. If it was not specified or does not
+// contains the namespace, fallback to the trace's namespace and podname.
+func getSeccompProfileNsName(
+	cli client.Client,
+	traceNs, traceOutputName, podname string,
+) (*SeccompProfileNsName, error) {
+	if traceOutputName != "" {
+		parts := strings.SplitN(traceOutputName, "/", 2)
+		if len(parts) == 2 {
+			// Use namespace and prefix-name provided by the user.
+			return &SeccompProfileNsName{
+				namespace:    parts[0],
+				name:         parts[1],
+				generateName: true,
+			}, nil
+		}
+
+		// Fallback to the trace's namespace and use prefix-name provided by the user.
+		return &SeccompProfileNsName{
+			namespace:    traceNs,
+			name:         traceOutputName,
+			generateName: true,
+		}, nil
+	}
+
+	// Fallback to the trace's namespace and podname but adding a counter
```

Comment:
Maybe this comment should go to line 270?

Reply:
Why do you think so? Is it because `adding a counter suffix in case there is already a profile with the podname name` is not applicable for the [`else`](https://github.com/kinvolk/inspektor-gadget/blob/6751cef84e32a5c840f9433aa9a18e0bd53fa3bc/pkg/gadgets/seccomp/gadget.go#L272-L275)?

Reply:
I wrote my comment before the addition of the `else`.
Thus, I thought we needed to maybe move this comment where the `SeccompProfileNsName` is instantiated.
But now with the `else`, I think it is better to let it here.